File: daily/2025/1863_Sum_of_All_Subset_XOR_Totals.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int total = 0;
    void subproblem(vector<int>& nums, int index, int curr){
        if (index == nums.size()){
            total += curr;
        }
        else{
            subproblem(nums, index+1, curr ^ nums[index]);
            subproblem(nums, index+1, curr);
        }

    }
    int subsetXORSum(vector<int>& nums) {
        subproblem(nums, 0, 0);
        return total;
    }
};

int main(){
    vector<int> nums = {1,3};
    // vector<int> nums = {5,1,6};
    // vector<int> nums = {3,4,5,6,7,8};
    Solution S;
    cout << S.subsetXORSum(nums) << endl;
    return 0;
}

File: daily_python/1863_Sum_of_All_Subset_XOR_Totals.py
import os
import sys
from typing import List

class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        total = 0
        for num in nums:
            total |= num
        return total << (len(nums) - 1)
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.subsetXORSum(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')